app/models/user.rb

```diff
@@ -1,4 +1,20 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
```

Comment:
In here you can setup a validation of the roles.
`validates_inclusion_of :roles, in: [ 'admin',  'author', 'visitor', 'subscriber']`
then in your unit test you can do
`it { is_expected.to validate_inclusion_of(:role).in_array(['admin',  'author', 'visitor', 'subscriber']) }`